src/pubsub.rs

```diff
@@ -274,45 +270,105 @@ impl PubSub {
             None => return,
         };
 
-        self.serialize_buf.clear();
-        match bincode::serialize_into(&mut self.serialize_buf, item) {
-            Ok(()) => tx_mapping.enqueue_bulk_bytes(&[(topic.as_ref(), &self.serialize_buf)]),
+        if self.serialize_bufs.is_empty() {
+            self.serialize_bufs.push(Vec::new());
+        }
+
+        self.serialize_bufs[0].clear();
+        match bincode::serialize_into(&mut self.serialize_bufs[0], item) {
+            Ok(()) => tx_mapping.enqueue_bulk_bytes(&[(topic.as_ref(), &self.serialize_bufs[0])]),
             Err(_) => return,
         };
     }
 
-    /// Dequeue a message from the broker, returning the topic and the message
-    /// in a newly-allocated Vec<u8>
-    pub fn dequeue_bytes(&mut self) -> Option<(&str, &[u8])> {
-        let rx_mapping = self.rx_mapping.as_mut()?;
-        
-        self.dequeue_topic.clear();
-        self.dequeue_buf.clear();
-        
-        let mut data = [(Some(&mut self.dequeue_topic), &mut self.dequeue_buf)];
-        let dequeued = rx_mapping.dequeue_bulk_bytes(&mut data, true);
-        
-        if dequeued != 1 {
-            return None;
+    /// Enqueue multiple messages to be sent to the broker under their respective topics.
+    /// 
+    /// Each message is serialized using bincode. The messages are provided as
+    /// tuples of (topic, item).
+    pub fn enqueue_bulk_type<S, T>(&mut self, items: &[(S, &T)])
+    where
+        S: AsRef<str>,
+        T: serde::Serialize,
+    {
+        let tx_mapping = match self.tx_mapping.as_mut() {
+            Some(m) => m,
+            None => return,
+        };
+
+        // resize buffer vec if needed
+        if self.serialize_bufs.len() < items.len() {
+            self.serialize_bufs.resize_with(items.len(), Vec::new);
+        }
+
+        for (i, (_, item)) in items.iter().enumerate() {
+            let buf = &mut self.serialize_bufs[i];
+            buf.clear();
+            if bincode::serialize_into(buf, *item).is_err() {
+                return;
+            }
+        }
+
+        let pairs: Vec<(&str, &[u8])> = items.iter()
```

Comment:
Not a huge fan of allocating a new vec here, don't know how we could prevent that tho